source/tests/pt/model/test_descriptor_sezm_cute_k1_structural_gate_vec4_sm80.py

```
@@ -0,0 +1,516 @@
+                logits = [
+                    _f32(rng.uniform(-4.0, 4.0)) for _ in range(2 * edge_count * 3 * 32)
+                ]
+                expected = _reference_forward(residual, y, logits, edge_count)
+                actual = residual.copy()
+                _vec4_forward_inplace_model(actual, y, logits, edge_count)
+                self.assertEqual(actual, expected)
+
+    def test_backward_dynamic_edges_and_block_tails(self):
+        for edge_count in DYNAMIC_EDGE_COUNTS:
+            with self.subTest(edge_count=edge_count):
+                rng = random.Random(20260722 + edge_count)
```

Ruff (0.16.1)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)